### DISCUSSION TITLE
Persist visible labels in log explorer

### DISCUSSION THREAD

**Discussion by mistic100** [grafana/grafana#56446] [Observability]
Hello

I use Grafana Explorer to consult logs stored on Loki. My applications publish ~20 different labels, not always relevant. Everytime I open the explorer I have to re-select which labels I want to see on the log line, and which are only visible when openning the log details.

It would greatly ease the exploitation to store the visible labels somewhere (user preferences ?) and re-apply the same setting the next time I open the explorer.

Here is an example of configuration I always apply :
![Capture d’écran du 2022-10-06 10-57-35](https://user-images.githubusercontent.com/41597/194269935-b682928a-423b-4d2a-a1ba-570c418cf69c.png)

**Comment by ifrost** [comment]
cc @grafana/observability-logs

**Comment by gabor** [comment]
thanks for the info @mistic100 ... i do wonder, how would this apply to other queries? like, would it be annoying when you look at some other queries in the same datasource? or is it like "whenever i use this datasource, these field-names i always want on the log-line"?

**Reply by mistic100** [reply]
@gabor I guess you mean if the datasource contains heterogenous data ? That's not my case. But I understand the concern and I have no answer as I didn't though about it.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
